### PR TITLE
Benchmark and Model Evaluation Metrics

### PR DESCRIPTION
**Pull request by sridhs21**
This PR adds performance tracking and evaluation metrics.

### Changes

**Performance Benchmarking** (new module: `benchmark.py`)
- Tracks hardware specifications (CPU, GPU, RAM, CUDA version)
- Monitors training performance: epoch timing, batch throughput, samples/second
- Records GPU memory usage (current and peak allocation)
- Exports benchmark data to JSON via `--benchmark` and `--benchmark-output` flags

**Model Evaluation Metrics** (new module: `eval_metrics.py`)
- Evaluates model on both training and validation sets post-training
- Computes standard metrics: Precision, Recall, F1 Score, IoU, Accuracy
- Provides confusion matrix breakdown (TP, FP, FN, TN) per-frame and globally
- Automatic overfitting detection by comparing train vs validation F1 scores
- Exports evaluation results to JSON via `--eval-output` flag

**Integration in `XPointMLTest.py`**
- Import benchmark and evaluation modules
- Pass benchmark tracker through training loop when enabled
- Run evaluation after training completes
- Generate formatted console output and JSON exports for both systems

### Benefits

- Monitor hardware utilization and training efficiency
- Quantify model accuracy with standard ML metrics
- Detect overfitting automatically
- Enable reproducible comparisons across runs via JSON exports

### CODE REVIEW

File: eval_metrics.py [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+    - False Negatives (FN): X-point pixels that were missed
+    - True Negatives (TN): Background pixels correctly identified
+    
+    Metrics calculated:
+    - Accuracy: (TP + TN) / (TP + TN + FP + FN)
+    - Precision: TP / (TP + FP)
+    - Recall: TP / (TP + FN)
+    - F1 Score: 2 * (Precision * Recall) / (Precision + Recall)
+    - IoU: TP / (TP + FP + FN)
+    """
```
Comment (cwsmith) [inline eval_metrics.py:31]: Great.  Thank you.

**Comment by cwsmith** [comment]
@sridhs21 Thank you.  Would you please rebase on main or merge main into this branch?

**Review by cwsmith — APPROVED** [review]
Looks great.  Thank you.
